src/core/builders/create-form-validator.ts

```diff
@@ -273,6 +275,20 @@ const getDependents = <Err>(
     }
   });
 
+const getParentsChain = <Err>(
+  desc: FieldDescriptor<any, Err>,
+  chain: FieldDescriptor<any, Err>[] = []
+): FieldDescriptor<any, Err>[] => {
+  const parent = impl(desc).__parent;
```

Comment:
maybe we can use the new `__parent` prop to improve `getRootArrayPath` function below?